extensions/typescript-language-features/src/extension.browser.ts

```diff
@@ -93,5 +97,31 @@ export function activate(
 
 	context.subscriptions.push(lazilyActivateClient(lazyClientHost, pluginManager, activeJsTsEditorTracker));
 
+	if (isWebAndHasSharedArrayBuffers()) {
+		await preload(logger); // TODO: Defer this and only do it when user has turned on this feature
+	}
+
 	return getExtensionApi(onCompletionAccepted.event, pluginManager);
 }
+
+async function preload(logger: Logger): Promise<void> {
+	const workspaceUri = vscode.workspace.workspaceFolders?.[0].uri;
+	if (!workspaceUri || workspaceUri.scheme !== 'vscode-vfs' || workspaceUri.authority !== 'github') {
+		return undefined;
+	}
+
+	try {
+		const remoteHubApi = await RemoteRepositories.getApi();
+		if (remoteHubApi.loadWorkspaceContents !== undefined) {
+			if (await remoteHubApi.loadWorkspaceContents(workspaceUri)) {
```

Comment:
@joyceerhl @dbaeumer @jrieken How does the `loadWorkspaceContents` API work? After calling this, do I need to switch to read from a different virtual file system? Or after `loadWorkspaceContents` completes does `vscode-vfs` transparently start using the downloaded data?

Also any steps for resting this locally on vscode-dev? Running on localhost with `SharedArrayBuffer` enabled, making this call results in:

![Screenshot 2023-01-18 at 4 35 03 PM](https://user-images.githubusercontent.com/12821956/213328194-2bbd5a48-839e-4c7c-831c-7ef856259ea3.png)

Reply:
No, you can read from the same URI. @joyceerhl will handle everything transparently.

When testing this are you adding the `?vscode-coi=` to the URI in the browser?

Reply:
What you are doing looks correct, just wait the call and everything should become faster. It might be that remote hub's trick isn't working locally, e.g that GH itself doesn't allow to be called from that origin.

Reply:
Thanks! Glad to hear no extra work is required

For debugging locally, I did add `vscode-coi=`. The full url I'm using is: 

```
http://localhost:3000/github/mjbvz/vscode-markdown-mermaid?vscode-quality=dev&vscode-log=trace&vscode-coi=
```